sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -49,6 +49,14 @@ class FileStreamSource(
     fs.makeQualified(new Path(path))  // can contains glob patterns
   }
 
+  private val optionsWithPartitionBasePath = if (!SparkHadoopUtil.get.isGlobPath(new Path(path))) {
```

Comment:
Also just a style thing but is this more direct?

```
private val optionsWithPartitionBasePath = sourceOptions.optionMapWithoutPath ++
  if (!SparkHadoopUtil.get.isGlobPath(new Path(path)) && options.contains("path") {
    Map("basePath" -> path)
  } else {
    Map()
  }
```

Not sure, just avoided some repetition but is about the same amount of code.